### DISCUSSION TITLE
Decode gzip/deflate automatically when mocking a proxied request

### DISCUSSION THREAD

**Discussion by jschroeter** [mockoon/mockoon#453] [Help] [answered]
Maybe I'm missing something, but I didn't figure out a way how Mockoon can decode gzipped response bodies automatically when adding a proxied request as a mock. It always shows the raw content, which isn't that helpful:
<img width="588" alt="mockoon-gzip" src="https://user-images.githubusercontent.com/487046/115041214-8a90d380-9ed2-11eb-9b62-0070a18e064b.png">

I worked around it by setting a proxy request header:
<img width="415" alt="mockoon-gzip-proxy-header" src="https://user-images.githubusercontent.com/487046/115043534-e2303e80-9ed4-11eb-822f-ed70fc55aaab.png">

**Comment by 255kb** [comment, marked as answer]
There is currently no gzip decoding in Mockoon and the body is forwarded "as-is" to the proxied server/API. However, there is one bug that has been fixed for the next release (due this week): the forwarded body was stringified instead of being forwarded as a regular Buffer. So, starting with v1.13.0, you should get in your proxied API the exact same content that entered Mockoon. (mockoon/commons-server#3)

I hope this answer your question.
If you would like Mockoon to be able to decode gzipped entering bodies, I would kindly ask you to [open an issue](https://github.com/mockoon/mockoon/issues/new?assignees=&labels=&template=Feature_request.md) on this precise topic.

**Reply by rubaiate** [reply]
I had same issue and implemented decoding gzipped bodies for my local usage and learn about mockoon. Like to push changes if you see feature is worthwhile.

**Reply by jschroeter** [reply]
Thanks for your replies and happy to hear, that @rubaiate made some progress already.
I opened an issue: https://github.com/mockoon/mockoon/issues/479

**Reply by 255kb** [reply]
@rubaiate You can surely contribute! Let's discuss a little bit more on #479 to check what changes are needed.
